### PR TITLE
New thread handling

### PR DESCRIPTION
**Pull request by qjgn**
Previously, the class RTProtocol owned a thread which handled the polling of new network data.
The thread shared data with the main thread without any thread synchronization. Bad things could happen.

The goal of this pull request is to provide safe data synchronization while maintaining compability with the old interface.

Streaming is handled by RTStreamThread which polls for new packets and updates an RTState instance.
RTState contains the state of all network objects.
Every fixed update RTState is copied to the main thread and is accessed through methods and properties on RTClient.

Connection status is also a part of the RTState, which means we can now connect and start streaming in a non blocking fashion.
I've added ~~two~~ one new method for this.

~~Task<bool> ConnectAsync( ..~~
~~Which returns a task which resolves as true if the connection was successful~~

```
void StartConnecting( ...
```
Which simply does not wait for a result.

With the new connect method comes the need for a more detailed connection status than ```bool RTClient::IsConnected()```
I've added the ```RTClient::ConnectionStatus``` property for this which returns an enum with three variants.
```
Disconnected
Connecting
Connected
```
This can be used by GUIs for example to hide buttons and change status when a connection is under way.

I've picked and updated to the the latest LTS version of Unity 2019.4.1f1 (LTS).

Edit: 
Removed the async connect method to keep compability with older versions of Unity

### CODE REVIEW

File: Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTGUI.cs [hunk 1]
```diff
@@ -4,7 +4,7 @@
 using UnityEngine;
 using UnityEditor;
 using System.Collections.Generic;
-using QTMRealTimeSDK.Data;
+using Assets.Qualisys.QTM_Unity_Realtime_Streaming.Helpers;
```
Comment (helmesjo) [inline Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTGUI.cs]: In this case I'd say we keep the old namespaces. `QTMRealtimeSDK` looks a lot cleaner and more intuitive.

File: Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTGUIingame.cs [hunk 2]
```diff
@@ -1,11 +1,9 @@
 ﻿// Unity SDK for Qualisys Track Manager. Copyright 2015-2018 Qualisys AB
 //
 using UnityEngine;
-using System.Collections;
 using QTMRealTimeSDK;
 using System.Collections.Generic;
-using System.Linq;
-using QTMRealTimeSDK.Data;
+using Assets.Qualisys.QTM_Unity_Realtime_Streaming.Helpers;
```
Comment (helmesjo) [inline Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTGUIingame.cs]: Same goes here. Keep the old namespace.

File: Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTState.cs [hunk 3]
```diff
@@ -0,0 +1,180 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using QTMRealTimeSDK.Data;
+using QTMRealTimeSDK.Settings;
+using QualisysRealTime.Unity;
+using UnityEngine;
+using GazeVector = QualisysRealTime.Unity.GazeVector;
+
+namespace Assets.Qualisys.QTM_Unity_Realtime_Streaming.Helpers
```
Comment (helmesjo) [inline Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTState.cs]: Use the old namespace.

File: Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTState.cs [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+            //    if (index == -1)
+            //    {
+            //        return new Skeleton(x);
+            //    }
+            //    else 
+            //    {
+            //        this.mSkeletons[index].CopyFrom(x);
+            //        return this.mSkeletons[index];
+            //    }
+            //}).ToList();
```
Comment (helmesjo) [inline Qualisys/QTM-Unity-Realtime-Streaming/Helpers/RTState.cs]: Remove this.

**Review by helmesjo — APPROVED** [review]
Given that performance is not negatively affected, and old projects work out of the box, this looks fine for me.

**Comment by qjgn** [comment]
Some code in this branch uses features which does not compile on Unity 2017 (.NET 35). It would not require a big change to keep the compability, however the RTClient::ConnectAsync will have to be removed since async things are not supported.

**Comment by qjgn** [comment]
Got OK from the team to merge this PR

**Pull request merged**